lib/chef/shell.rb

```diff
@@ -295,18 +296,19 @@ def parse_opts
     private
 
     def config_file_for_shell_mode(environment)
+      dot_chef_dir = Chef::Util::PathHelper.home('.chef')
       if config[:config_file]
         config[:config_file]
-      elsif environment && ENV['HOME']
+      elsif environment && dot_chef_dir
```

Comment:
I think `elsif environment` here is the right thing to preserve behavior.  I'll make comments on any other sections if I think they should be changed too.